# Population decline analysis

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
pop= pd.read_csv("Total Births, Deaths and Marriages Registered.csv")
pop.head()

Statistic Label Quarter  State    UNIT   VALUE
0  Marriages Registered  1960Q1  State  Number  2987.0
1  Marriages Registered  1960Q2  State  Number  3501.0
2  Marriages Registered  1960Q3  State  Number  5900.0
3  Marriages Registered  1960Q4  State  Number  3106.0
4  Marriages Registered  1961Q1  State  Number  2179.0

In [14]:
pop.shape

(4554, 12)

## Note rationale for seeking domain knowledge and findings
In other to explore population declin hypothesis in ireland i consulted with domain knowledgible professionals
Populkation of married couples in ireland does not have a direct corrolation wirth population declin or increase according to domain knowledge. Because not all married people decide to have offsprings, some adopt and some where unable to have offsprings for biological r3easons. Thus is why i decided to drip the marriage column in the popularion projevt. 

In [3]:
pop['Married'] = pop['VALUE'][pop['Statistic Label']=='Marriages Registered']

In [4]:
pop.head()

Statistic Label Quarter  State    UNIT   VALUE  Married
0  Marriages Registered  1960Q1  State  Number  2987.0   2987.0
1  Marriages Registered  1960Q2  State  Number  3501.0   3501.0
2  Marriages Registered  1960Q3  State  Number  5900.0   5900.0
3  Marriages Registered  1960Q4  State  Number  3106.0   3106.0
4  Marriages Registered  1961Q1  State  Number  2179.0   2179.0

In [29]:
pop['Statistic Label'].unique()

array(['Marriages Registered',
       'Marriage Rate Registered per 1000 Estimated Population',
       'Deaths Registered',
       'Death Rate Registered per 1000 Estimated Population',
       'Deaths of Infants under 1 Year Registered per 1000 Births',
       'Births Registered',
       'Birth Rate Registered per 1000 Estimated Population',
       'Opposite sex marriages', 'Same sex marriages',
       'Civil partnerships', 'Births registered that were first births',
       'Births registered outside marriage', 'Opposite sex marriage rate',
       'Same sex marriage rate', 'Civil partnership marriage rate',
       'Percentage of first births registered',
       'Percentage of births registered outside marriage',
       'Average age of mothers giving birth'], dtype=object)

As shown above in the pop dataframe(df), there are by deault 4 columns and 4554 rows. However, I used 'unique()' method  on Statistic Label column to see the unique observations in it, this unique values have their corresponding values. In other to explore each unique oberservation and its corresponding values, i need to extract the unique values into columns. Frthermore the values values are observaved for 4 quarters in a year. 

In [17]:
pop['Reg_Death'] = pop['VALUE'][pop['Statistic Label']=='Deaths Registered']

In [18]:
pop['Rate_Death/1000'] = pop['VALUE'][pop['Statistic Label']=='Death Rate Registered per 1000 Estimated Population']

In [19]:
pop['Reg_inf_Death/1000'] = pop['VALUE'][pop['Statistic Label']=='Deaths of Infants under 1 Year Registered per 1000 Births']

In [20]:
pop['Reg_Birth'] = pop['VALUE'][pop['Statistic Label']=='Births Registered']

In [21]:
pop['B_outsideMarriage'] = pop['VALUE'][pop['Statistic Label']=='Percentage of births registered outside marriage']

In [22]:
pop['Mothers_age'] = pop['VALUE'][pop['Statistic Label']== 'Average age of mothers giving birth']

In [23]:
pop['CV_P_M'] = pop['VALUE'][pop['Statistic Label']== 'Civil partnership marriage rate']

In [24]:
pop['Rate_marriage/1000'] = pop['VALUE'][pop['Statistic Label']== 'Marriage Rate Registered per 1000 Estimated Population']

In [25]:
pop.head()

Statistic Label Quarter  State    UNIT   VALUE  Married  Death  Birth  \
0  Marriages Registered  1960Q1  State  Number  2987.0   2987.0    NaN    NaN   
1  Marriages Registered  1960Q2  State  Number  3501.0   3501.0    NaN    NaN   
2  Marriages Registered  1960Q3  State  Number  5900.0   5900.0    NaN    NaN   
3  Marriages Registered  1960Q4  State  Number  3106.0   3106.0    NaN    NaN   
4  Marriages Registered  1961Q1  State  Number  2179.0   2179.0    NaN    NaN   

   B_outsideMarriage  Reg_Death/1000  Reg_inf_Death/1000  Mothers_age  CV_P_M  \
0                NaN             NaN                 NaN          NaN     NaN   
1                NaN             NaN                 NaN          NaN     NaN   
2                NaN             NaN                 NaN          NaN     NaN   
3                NaN             NaN                 NaN          NaN     NaN   
4                NaN             NaN                 NaN          NaN     NaN   

   Reg_Death  Rate_Death/1000  Reg_Birth  Rate_marriage/1000  
0        NaN              NaN        NaN                 NaN  
1        NaN              NaN        NaN                 NaN  
2        NaN              NaN        NaN                 NaN  
3        NaN              NaN        NaN                 NaN  
4        NaN              NaN        NaN                 NaN

In [32]:
pop.tail()

Statistic Label Quarter  State    UNIT  VALUE  \
4549  Average age of mothers giving birth  2022Q1  State  Number   33.3   
4550  Average age of mothers giving birth  2022Q2  State  Number   33.2   
4551  Average age of mothers giving birth  2022Q3  State  Number   33.2   
4552  Average age of mothers giving birth  2022Q4  State  Number   33.2   
4553  Average age of mothers giving birth  2023Q1  State  Number   33.2   

      Married  Death  Birth  B_outsideMarriage  Reg_Death/1000  \
4549      NaN    NaN    NaN                NaN             NaN   
4550      NaN    NaN    NaN                NaN             NaN   
4551      NaN    NaN    NaN                NaN             NaN   
4552      NaN    NaN    NaN                NaN             NaN   
4553      NaN    NaN    NaN                NaN             NaN   

      Reg_inf_Death/1000  Mothers_age  CV_P_M  Reg_Death  Rate_Death/1000  \
4549                 NaN         33.3     NaN        NaN              NaN   
4550                 NaN         33.2     NaN        NaN              NaN   
4551                 NaN         33.2     NaN        NaN              NaN   
4552                 NaN         33.2     NaN        NaN              NaN   
4553                 NaN         33.2     NaN        NaN              NaN   

      Reg_Birth  Rate_marriage/1000  
4549        NaN                 NaN  
4550        NaN                 NaN  
4551        NaN                 NaN  
4552        NaN                 NaN  
4553        NaN                 NaN

In [30]:
pop['Quarter'].unique()

array(['1960Q1', '1960Q2', '1960Q3', '1960Q4', '1961Q1', '1961Q2',
       '1961Q3', '1961Q4', '1962Q1', '1962Q2', '1962Q3', '1962Q4',
       '1963Q1', '1963Q2', '1963Q3', '1963Q4', '1964Q1', '1964Q2',
       '1964Q3', '1964Q4', '1965Q1', '1965Q2', '1965Q3', '1965Q4',
       '1966Q1', '1966Q2', '1966Q3', '1966Q4', '1967Q1', '1967Q2',
       '1967Q3', '1967Q4', '1968Q1', '1968Q2', '1968Q3', '1968Q4',
       '1969Q1', '1969Q2', '1969Q3', '1969Q4', '1970Q1', '1970Q2',
       '1970Q3', '1970Q4', '1971Q1', '1971Q2', '1971Q3', '1971Q4',
       '1972Q1', '1972Q2', '1972Q3', '1972Q4', '1973Q1', '1973Q2',
       '1973Q3', '1973Q4', '1974Q1', '1974Q2', '1974Q3', '1974Q4',
       '1975Q1', '1975Q2', '1975Q3', '1975Q4', '1976Q1', '1976Q2',
       '1976Q3', '1976Q4', '1977Q1', '1977Q2', '1977Q3', '1977Q4',
       '1978Q1', '1978Q2', '1978Q3', '1978Q4', '1979Q1', '1979Q2',
       '1979Q3', '1979Q4', '1980Q1', '1980Q2', '1980Q3', '1980Q4',
       '1981Q1', '1981Q2', '1981Q3', '1981Q4', '1982Q1', '1982

In [31]:
pop['Quarter'].shape

(4554,)

In [26]:
pop['Death'].isnull().sum()

4301

In [27]:
pop['Death'].shape

(4554,)

In [28]:
pop.isnull().sum()

Statistic Label          0
Quarter                  0
State                    0
UNIT                     0
VALUE                 2728
Married               4301
Death                 4301
Birth                 4301
B_outsideMarriage     4549
Reg_Death/1000        4301
Reg_inf_Death/1000    4301
Mothers_age           4549
CV_P_M                4549
Reg_Death             4301
Rate_Death/1000       4301
Reg_Birth             4301
Rate_marriage/1000    4301
dtype: int64

As shown above, though we have been able to extract the values of unique o=bservations in the statistic label, we have created understandablly enomous mising values. This is because where there are no observations, pandas will authomatically repkace the rows as missing in other to balance with the shape of the data. 

## Percentage of null values
This operation is used to calculate the proportion of missing values with values that are recorded in 

In [ ]:
df.isnull().mean()*100 # Porcentage of missing data